### PR TITLE
Revert with a readable msg if relayers data length == 0

### LINKED ISSUE #230: Require `_data.length` > 0
https://github.com/darwinia-network/darwinia-messages-sol/blob/28ec79482a0d5ef500fe1ce066c51165f4efe75c/contracts/periphery/contracts/s2s/types/CommonTypes.sol#L81

### PR DESCRIPTION
**Pull request by wuminzhe**
Fix #230

### CODE REVIEW

File: contracts/utils/contracts/ScaleCodec.sol [hunk 1]
```diff
@@ -19,6 +19,7 @@ library ScaleCodec {
         pure
         returns (uint256 v, uint8 m)
     {
+        require(data.length > 0, "Compact data cannot be empty");
```
Comment (hujw77) [inline contracts/utils/contracts/ScaleCodec.sol:22]: It is safe, but is a redundant check.
If the decode type is `uintN`, data.length > 0.

Comment (wuminzhe) [reply contracts/utils/contracts/ScaleCodec.sol:22]: > If the decode type is uintN, data.length > 0.
what's this mean?

Comment (hujw77) [reply contracts/utils/contracts/ScaleCodec.sol:22]:
```
bytes data = scalec.encode(uintN);
data.length > 0
```

Comment (hujw77) [reply contracts/utils/contracts/ScaleCodec.sol:22]: If the data is uintN type, data.length > 0.

Comment (hujw77) [reply contracts/utils/contracts/ScaleCodec.sol:22]: If you want do a unitN scalec decode. the check is redundant.
This issue occur root case is that decode a Non UintN scale type.

**Review by hujw77 — APPROVED** [review]
LGM

**Pull request merged**